cabot/cabotapp/models.py

```diff
@@ -934,6 +940,39 @@ def _run(self):
         return result
 
 
+class TCPStatusCheck(StatusCheck):
+
+    @property
+    def check_category(self):
+        return "TCP Check"
+
+    @property
+    def description(self):
+        return 'Monitoring {0}:{1}'.format(self.address, self.port)
+
+    address = models.TextField()
+    port = models.PositiveIntegerField()
+
+    update_url = 'update-tcp-check'
+
+    icon = 'glyphicon glyphicon-text-height'
+
+    def _run(self):
+        result = StatusCheckResult(check=self)
+
+        tcp_process = subprocess.Popen("nc -G 12 -vz " + self.address + " " + str(self.port),
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.STDOUT,
```

Comment:
From what I can tell nc only writes to stderr for errors? So we can just pipe the stderr output instead of redirecting to stdout and then do 
`result.error = tcp_process.stderr.read()`